### PR TITLE
Tyk Chart 2.2 release

### PR DESCRIPTION
**Pull request by carolwkc**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

---

### Preview Link

<br>

https://deploy-preview-5719--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-charts/release-notes/version-2.2/

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labeled** the PR

___

### **PR Type**
Documentation, Enhancement

___

### **Description**
- Added comprehensive release notes for Tyk Charts version 2.2, detailing new features, enhancements, and fixed issues.
- Introduced new configurations for audit logs and Open Policy Agent (OPA) in Tyk Dashboard.
- Added support for OpenTelemetry and customizable liveness and readiness probes for Tyk Gateway.
- Updated the documentation menu to include a link to the Tyk Charts 2.2 release notes.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>version-2.2.md</strong><dd><code>Added release notes for Tyk Charts version 2.2</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.2.md

<li>Added release notes for Tyk Charts version 2.2.<br> <li> Documented new features, enhancements, and fixed issues.<br> <li> Included upgrade instructions and download links.<br> <li> Provided a detailed changelog with added, changed, and fixed sections.<br> <br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5719/files#diff-e46c941a6fbfd80555ffc24cb56c3116fb367873705771df64551932eb5f543e">+290/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Updated menu with Tyk Charts 2.2 release notes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml

- Added menu entry for Tyk Charts 2.2 release notes.

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5719/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>tyk-control-plane-chart.md</strong><dd><code>Added audit log and OPA configurations for Dashboard</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md

<li>Added audit log configurations for Tyk Dashboard.<br> <li> Introduced OPA configurations for Tyk Dashboard.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5719/files#diff-1bb3ae7ce6b5aec344682558fe28a0e91ef5504473fbed884ae4651089784cf7">+19/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>tyk-data-plane-chart.md</strong><dd><code>Added OpenTelemetry and probes configuration for Gateway</code>&nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-data-plane-chart.md

<li>Added OpenTelemetry configuration for Gateway.<br> <li> Added liveness and readiness probes configuration for Gateway.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5719/files#diff-da211afa42cedf90ce80f9d59419356f09f0a76d37f56788b523bdf02b198e4f">+12/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>tyk-oss-chart.md</strong><dd><code>Added OpenTelemetry and probes configuration for Gateway</code>&nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-oss-chart.md

<li>Added OpenTelemetry configuration for Gateway.<br> <li> Added liveness and readiness probes configuration for Gateway.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5719/files#diff-53fd570dd4fbaa9b39eca159ac973030b7f820da118b68df42df490432741dd5">+12/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>tyk-stack-chart.md</strong><dd><code>Added probes, audit log, and OPA configurations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md

<li>Added liveness and readiness probes configuration for Gateway.<br> <li> Added audit log configurations for Tyk Dashboard.<br> <li> Introduced OPA configurations for Tyk Dashboard.<br>

</details>

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5719/files#diff-f6e45bbf530fa763d587dec798584c8eb6294229ef353d3212ab01bf6b0a1d7c">+24/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.2.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+#### Changed
+
+<ul>
+
+<li>
+<details>
+<summary>Updated default versions of Tyk components</summary>
+
+ Tyk Charts 2.1 will install the following Tyk component versions by default.
```
Comment (sharadregoti) [inline tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.2.md]:
```suggestion
 Tyk Charts 2.2 will install the following Tyk component versions by default.
```

File: tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.2.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+Support for configuring the Tyk Gateway logging level and format through new fields under `.Values.gateway.log` in the Helm chart values.yaml.
+
+This enhancement enables fine-tuned control over logging behavior directly from the Helm chart, simplifying deployment customization.
+</details>
+</li>
+
+<li>
+<details>
+<summary>Gateway: Customizable HPA behavior</summary>
+Users can now define custom HPA behavior settings directly in the Helm values file via a new field a new field: `.Values.gateway.autoscaling.behavior`.
```
Comment (sharadregoti) [inline tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.2.md:184]:
```suggestion

Users can now define custom HPA behavior settings directly in the Helm values file via a new field a new field: `.Values.gateway.autoscaling.behavior`.
```
Added a new line, for proper rendering of code block

**Review by sharadregoti — CHANGES_REQUESTED** [review]

**Review by sharadregoti — APPROVED** [review]

**Review by sharadregoti — APPROVED** [review]

File: tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.2.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+## Support Lifetime
+<!-- Required. replace X.Y with this release and set the correct quarter of the year -->
+Our minor releases are supported until our next minor comes out. 
+
+---
+
+## 2.2.0 Release Notes
+
+### Release Date DD Mon YYYY <<update>>
```
Comment (sharadregoti) [inline tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.2.md]:
```suggestion
### Release Date 09 December 2024
```
TODO: Change `date` when merging the PR.

File: tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.2.md [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+The `externalTrafficPolicy` field is now correctly set under the spec section instead of selectors.
+This fix ensures proper functionality and alignment with Kubernetes service configuration requirements.
+</details>
+</li>
+
+<li>
+<details>
+<summary>Operator: Liveness and readiness probes failed due to deprecated ControllerManagerConfig CRD</summary>
+
+Tyk Operator manifests used ControllerManagerConfig CRD to configure Operator Manager. However, the ControllerManagerConfig CRD has been deprecated in Tyk Operator v1.0 when we upgraded controller-runtime. Even though, operator runtime has default values for the configurations defined in CRD, deprecated CRD creates confusion for customers while deploying Tyk Operator.
```
Comment (komalsukhani) [inline tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.2.md]: @caroltyk This change is not included. Am I right @buraksekili ?

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 5]
```diff
@@ -664,6 +664,25 @@ Optional Steps, if needed:
 - If the `tyk-bootstrap` chart is used to bootstrap the Tyk Dashboard, ensure that bootstrap app can validate certificate of Tyk Dashboard or enable `insecureSkipVerify` in the `tyk-bootstrap` chart.
 - If the Tyk Gateway connects to the Tyk Dashboard, confirm that the Tyk Gateway has appropriate certificates for connecting to the Tyk Dashboard
 
+#### Audit Log Configurations
+
+You can manage audit logging for Tyk Dashboard via `auditLogs`:
+
+- `auditLogs.enabled`: Enables or disables audit logging. It sets corresponding Dashboard environment variable `TYK_DB_AUDIT_ENABLED`. Enabled by default.
+- `auditLogs.type`: Specifies the storage type for audit logs (`db` or `file`). It sets corresponding Dashboard environment variable `TYK_DB_AUDIT_TYPE`. Set to `file` by default.
```
Comment (komalsukhani) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]:
```suggestion
- `auditLogs.type`: Specifies the storage type for audit logs (`db` or `file`). It sets corresponding Dashboard environment variable `TYK_DB_AUDIT_STORETYPE`. Set to `file` by default.
```

File: tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md [hunk 6]
```diff
@@ -760,6 +765,25 @@ Optional Steps, if needed:
 - If the `tyk-bootstrap` chart is used to bootstrap the Tyk Dashboard, ensure that it has certificates to send requests to the Tyk Dashboard or enable `insecureSkipVerify` in the `tyk-bootstrap` chart.
 - If the Tyk Gateway connects to the Tyk Dashboard, confirm that the Tyk Gateway has appropriate certificates for connecting to the Tyk Dashboard
 
+#### Audit Log Configurations
+
+You can manage audit logging for Tyk Dashboard via `auditLogs`:
+
+- `auditLogs.enabled`: Enables or disables audit logging. It sets corresponding Dashboard environment variable `TYK_DB_AUDIT_ENABLED`. Enabled by default.
+- `auditLogs.type`: Specifies the storage type for audit logs (`db` or `file`). It sets corresponding Dashboard environment variable `TYK_DB_AUDIT_TYPE`. Set to `file` by default.
```
Comment (komalsukhani) [inline tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md]:
```suggestion
- `auditLogs.type`: Specifies the storage type for audit logs (`db` or `file`). It sets corresponding Dashboard environment variable `TYK_DB_AUDIT_STORETYPE`. Set to `file` by default.
```

File: tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md [hunk 7]
```diff
@@ -760,6 +765,25 @@ Optional Steps, if needed:
 - If the `tyk-bootstrap` chart is used to bootstrap the Tyk Dashboard, ensure that it has certificates to send requests to the Tyk Dashboard or enable `insecureSkipVerify` in the `tyk-bootstrap` chart.
 - If the Tyk Gateway connects to the Tyk Dashboard, confirm that the Tyk Gateway has appropriate certificates for connecting to the Tyk Dashboard
 
+#### Audit Log Configurations
+
+You can manage audit logging for Tyk Dashboard via `auditLogs`:
+
+- `auditLogs.enabled`: Enables or disables audit logging. It sets corresponding Dashboard environment variable `TYK_DB_AUDIT_ENABLED`. Enabled by default.
```
Comment (komalsukhani) [inline tyk-docs/content/product-stack/tyk-charts/tyk-stack-chart.md]:
```suggestion
- `auditLogs.enabled`: Enables or disables audit logging. It sets corresponding Dashboard environment variable `TYK_DB_AUDIT_ENABLED`. Disabled by default.
```

File: tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md [hunk 8]
```diff
@@ -664,6 +664,25 @@ Optional Steps, if needed:
 - If the `tyk-bootstrap` chart is used to bootstrap the Tyk Dashboard, ensure that bootstrap app can validate certificate of Tyk Dashboard or enable `insecureSkipVerify` in the `tyk-bootstrap` chart.
 - If the Tyk Gateway connects to the Tyk Dashboard, confirm that the Tyk Gateway has appropriate certificates for connecting to the Tyk Dashboard
 
+#### Audit Log Configurations
+
+You can manage audit logging for Tyk Dashboard via `auditLogs`:
+
+- `auditLogs.enabled`: Enables or disables audit logging. It sets corresponding Dashboard environment variable `TYK_DB_AUDIT_ENABLED`. Enabled by default.
```
Comment (komalsukhani) [inline tyk-docs/content/product-stack/tyk-charts/tyk-control-plane-chart.md]:
```suggestion
- `auditLogs.enabled`: Enables or disables audit logging. It sets corresponding Dashboard environment variable `TYK_DB_AUDIT_ENABLED`. Disabled by default.
```

**Review by komalsukhani — CHANGES_REQUESTED** [review]

File: tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.2.md (hunk 4 above)
Comment (buraksekili) [reply tyk-docs/content/product-stack/tyk-charts/release-notes/version-2.2.md]: yes, you are right @komalsukhani. the change has not been merged yet.

**Review by sharadregoti — APPROVED** [review]

**Review by komalsukhani — APPROVED** [review]

**Pull request merged**